scheds/rust/scx_layered/src/main.rs

```diff
@@ -1030,16 +1048,43 @@ impl Layer {
             }
         }
 
+        let mut nodes = topo.nodes().iter().collect::<Vec<_>>().clone();
+        let num_nodes = nodes.len();
+        nodes.rotate_left(if idx <= num_nodes { idx } else { idx % num_nodes });
```

Comment:
This is the main logic, it makes use of [`rotate_left`](https://doc.rust-lang.org/nightly/std/vec/struct.Vec.html#method.rotate_left), which annoying will panic if the rotate size is too large. To get around that it mods by the number of nodes, llcs, or cores if the number of layers is larger than any of those values.